src/main/java/com/firebolt/jdbc/type/array/SqlArrayUtil.java

```diff
@@ -221,4 +225,31 @@ private static Object[] toObjectArray(Object array) {
 		}
 		return ret;
 	}
+
+	public static String byteArrayToHexString(byte[] bytes, boolean separateEachByte) {
+		if (bytes == null)  {
+			return null;
+		}
+		ByteBuffer buffer = ByteBuffer.wrap(bytes);
+		String separator = separateEachByte ? BYTE_ARRAY_PREFIX : "";
+		return BYTE_ARRAY_PREFIX + Stream.generate(buffer::get).limit(buffer.capacity()).map(Integer::toHexString).collect(joining(separator));
+	}
+
+	@SuppressWarnings("java:S1168") // we have to return null here
+	public static byte[] hexStringToByteArray(String str) {
+		if (str == null) {
+			return null;
+		}
+		if (!str.startsWith(BYTE_ARRAY_PREFIX)) {
+			return str.getBytes();
```

Comment:
Reading up on `getBytes`, we probably should pass UTF-8 charset to it otherwise it uses a system-default one, which makes the decoding behaviour system-dependent.
https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#getBytes-java.nio.charset.Charset-

Reply:
Funny. I was thing about this while writing this code but than forgot. Fixed.